Description: fixes for GCC 4.7
 It seems this version of GCC has stricter requirements to some C++ scopes
Author: Damyan Ivanov <dmn@debian.org>
Forwarded: no

--- a/src/common/classes/array.h
+++ b/src/common/classes/array.h
@@ -347,7 +347,7 @@ public:
 	size_t add(const Value& item) {
 	    size_t pos;
   	    find(KeyOfValue::generate(this, item), pos);
-		insert(pos, item);
+		this->insert(pos, item);
 		return pos;
 	}
 };
--- a/src/common/classes/vector.h
+++ b/src/common/classes/vector.h
@@ -135,7 +135,7 @@ public:
 	size_t add(const Value& item) {
 	    size_t pos;
   	    find(KeyOfValue::generate(this, item), pos);
-		insert(pos, item);
+		this->insert(pos, item);
 		return pos;
 	}
 };
--- a/src/common/classes/stack.h
+++ b/src/common/classes/stack.h
@@ -48,7 +48,7 @@ namespace Firebird {
 			Entry(Object e, Entry* stk) 
 				: inherited(), next(stk) 
 			{
-				add(e);
+				this->add(e);
 			}
 			
 			Entry(Entry* stk) : inherited(), next(stk) { }
@@ -62,7 +62,7 @@ namespace Firebird {
 			{
 				if (inherited::getCount() < this->getCapacity())
 				{
-					add(e);
+					this->add(e);
 					return this;
 				}
 				Entry* newEntry = FB_NEW(p) Entry(e, this);
--- a/src/jrd/Collation.cpp
+++ b/src/jrd/Collation.cpp
@@ -191,6 +191,15 @@ bool MATCHESNAME(Jrd::thread_db* tdbb, J
 
 
 template <class SLEUTHTYPE>
+static bool SLEUTH_AUX(
+						  Jrd::TextType* obj,
+						  USHORT flags,
+						  const SLEUTHTYPE* search,
+						  const SLEUTHTYPE* end_search,
+	const SLEUTHTYPE* match, const SLEUTHTYPE* end_match);
+
+
+template <class SLEUTHTYPE>
 bool SLEUTHNAME(Jrd::thread_db* tdbb_dummy, Jrd::TextType* obj, USHORT flags,
 				const SLEUTHTYPE* search, SLONG search_len,
 				const SLEUTHTYPE* match, SLONG match_len)
@@ -343,6 +352,15 @@ ULONG SLEUTH_MERGE_NAME(Jrd::thread_db*
 
 
 template <class SLEUTHTYPE>
+static bool SLEUTH_CLASS_NAME(
+								 Jrd::TextType* obj,
+								 USHORT flags,
+								 const SLEUTHTYPE* char_class,
+								 const SLEUTHTYPE* const end_class,
+								 SLEUTHTYPE character);
+
+
+template <class SLEUTHTYPE>
 static bool SLEUTH_AUX(
 						  Jrd::TextType* obj,
 						  USHORT flags,
